File: Week1/CountingSort/src/main/java/MyCode.java
package main.java;

import java.util.ArrayList;
import java.util.List;

public class MyCode {
    public static List<Integer> countingSort(List<Integer> arr) {
        List<Integer> results = new ArrayList<>(100);
        for (int i = 0; i< 100; i++){
            results.add(0);
        }
        arr.stream().forEach(
                i-> results.set(i,results.get(i)+1)
        );
        return results;
    }
}

File: Week1/CountingSort/src/main/kotlin/MyCode.kt
package main.kotlin

fun countingSort(arr: Array<Int>): Array<Int>{
    val results = Array<Int>(100,{0})

    for (i in results.indices){
        for(j in arr.indices){
            if( i == arr[j]) results[i]++
        }
    }
    return results

}
fun main(args: Array<String>) {
    val n = readLine()!!.trim().toInt()

    val arr = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()

    val result = countingSort(arr)

    println(result.joinToString(" "))
}